x/mongo/driver/topology/topology_options.go

```diff
@@ -190,6 +190,7 @@ func WithConnString(fn func(connstring.ConnString) connstring.ConnString) Option
 					AppName:       cs.AppName,
 					Authenticator: authenticator,
 					Compressors:   cs.Compressors,
+					LoadBalanced:  cs.LoadBalancedSet && cs.LoadBalanced,
```

Comment:
Hmm interesting; were the tests failing without the changes to this file?

Reply:
Sorry for the slow response!

Yes. Because the initial DNS seedlist discovery spec test runner calls [`Topology.WithConnString`](https://github.com/mongodb/mongo-go-driver/blob/a2b9adad561b3ddeb9b54d911b2dc6473e9caa51/mongo/integration/initial_dns_seedlist_discovery_test.go#L114) instead of creating a `Client`, the logic for building a `topology.Option` from a `connstring.ConnString` is completely distinct from the logic used for a `Client` in [`Client.configure`](https://github.com/mongodb/mongo-go-driver/blob/a2b9adad561b3ddeb9b54d911b2dc6473e9caa51/mongo/client.go#L350). Without setting `LoadBalanced` correctly in the auth and no-auth handshakers, the test fails with errors like
```
initial_dns_seedlist_discovery_test.go:119: error finding host "localhost.test.build.10gen.cc:8000": connection() error occurred during connection handshake: (LoadBalancerSupportMismatch) The server is being accessed through a load balancer, but this driver does not have load balancing enabled
```
Fortunately `Topology.WithConnString()` appears to only be used in tests, but we should consider finding a way to remove it because it doesn't always behave the same as `Client.configure`.